iis-wf/iis-wf-export-actionmanager/src/test/java/eu/dnetlib/iis/wf/export/actionmanager/sequencefile/FieldAccessor.java

```diff
@@ -0,0 +1,143 @@
+
+	private static final char DECODER_PREFIX = '$';
+
+	private static final String GETTER = "get";
+
+	private Map<String, FieldDecoder> decoders = new HashMap<String, FieldDecoder>();
+
+	/**
+	 * Registers field decoder.
+	 * 
+	 * @param field
+	 * @param decoder
+	 */
+	public void registerDecoder(String field, FieldDecoder decoder) {
+		decoders.put(field, decoder);
+	}
+
+	/**
+	 * Extracts value from given object based on fieldPath.
+	 * @param fieldPath field names acessible vie getter methods, nested fields are separated with dots
+	 * @param object object value should be extracted from
+	 * @return extracted value
+	 * @throws Exception
+	 */
+	public Object getValue(String fieldPath, Object object) throws Exception {
+		try {
+			if (object != null && fieldPath != null) {
+				Object currentObject = object;
+				String[] tokens = StringUtils.split(fieldPath, FIELD_PATH_SEPARATOR);
+				for (String token : tokens) {
+					if (token.charAt(0) == DECODER_PREFIX) {
+						String fieldName = token.substring(1);
+						FieldDecoder decoder = decoders.get(fieldName);
+						if (decoder == null) {
+							throw new Exception("no decoder registered for " + token);
+						}
+						Object fieldValue = getFieldValue(fieldName, currentObject);
+						if (decoder.canHandle(fieldValue)) {
+							currentObject = decoder.decode(fieldValue);
+						} else {
+							throw new Exception(fieldValue!=null?
+									decoder.getClass().getName() + " decoder is not capable of handling object: " + 
+									fieldValue.toString() + ", which is " + fieldValue.getClass().getName() + " class instance"
+									:decoder.getClass().getName() + " decoder is not capable of handling null object");
+						}
+					} else if (token.indexOf('[')>0) {
+						String fieldName = token.substring(0, token.indexOf('['));
+						currentObject = getFieldValue(fieldName, currentObject);
+						if (currentObject!=null) {
+							 if (currentObject.getClass().isArray()) {
+								 currentObject = Array.get(currentObject, getArrayPositionFromToken(token));	 
+							 } else if (List.class.isAssignableFrom(currentObject.getClass())) {
+								 currentObject = ((List<?>)currentObject).get(getArrayPositionFromToken(token));
+							 } else {
+								 throw new Exception("unable to extract array or list element "
+											+ "based on field path: " + fieldPath + ", "
+											+ "object is neither an array nor list:" + 
+											currentObject.getClass().getName());
+							 }
+						} else {
+							throw new Exception("unable to extract array or list element "
+									+ "based on field path: " + fieldPath + ", "
+									+ "object is null!");
+						}
+					} else {
+						currentObject = getFieldValue(token, currentObject);
+					}
+				}
+				return currentObject;
+			} else {
+				return null;
+			}
+		} catch (Exception e) {
+//			we are extremely verbose to make log analysis process easier 
+			throw new Exception("unable to resolve field path '" + fieldPath 
```

Comment:
Finally I decided to introduce `FieldAccessorException` here and in all the places above. 

At first I didn't find justification for this (`TestingConsumer` treats all exceptions in the same way) but... it is always better to handle specific exceptions in upper layer, whatever it does. In future it can be utilized by other consumers which could do more with specific exception.